zrml/global-disputes/src/lib.rs

```diff
@@ -221,6 +225,9 @@ mod pallet {
         ) -> DispatchResultWithPostInfo {
             let owner = ensure_signed(origin)?;
 
+            let market = T::MarketCommons::market(&market_id)?;
+            ensure!(market.matches_outcome_report(&outcome), Error::<T>::OutcomeMismatch);
```

Comment:
It would make more sense to me to put this into `push_vote_outcome`. If you call `push_vote_outcome` from some other function and forget to verify that the outcome types match, then we'll have another bug.

Reply:
`add_vote_outcome` is different to `push_voting_outcome`. `add_vote_outcome` is controlled by a user. So, I would be in favour of blocking a user to put in a combinatoric outcome instead of a scalar outcome.

Reply:
But `add_vote_outcome` is implemented using `push_voting_outcome`. That's why I suggested it. You have a clean API, i.e. other pallets can't accidentally push incorrect outcomes, and neither can users.

Reply:
now decoupled https://github.com/zeitgeistpm/zeitgeist/blob/6611b2540956a4487dddc913f849acf8fb64279b/zrml/global-disputes/src/lib.rs#L263-L267

Reply:
after this change I suspect time complexity of this extrinsic has changed, and perhaps weight computation is also changed.

Reply:
Yeah, that's true.

Reply:
Not sure what was done to resolve this problem.

Reply:
It's just I said above. `add_vote_outcome` did use `push_vote_outcome`, but now it's decoupled, so the logic of `add_vote_outcome` is now independent from `push_vote_outcome` with this

https://github.com/zeitgeistpm/zeitgeist/blob/6611b2540956a4487dddc913f849acf8fb64279b/zrml/global-disputes/src/lib.rs#L263-L267